confluent_kafka/schema_registry/json_schema.py

```diff
@@ -0,0 +1,261 @@
+from io import BytesIO
+
+import json
+import struct
+
+from jsonschema import validate, ValidationError
+
+from confluent_kafka.schema_registry import (_MAGIC_BYTE,
+                                             Schema,
+                                             topic_subject_name_strategy)
+from confluent_kafka.serialization import SerializationError, Serializer, Deserializer
+
+
+class _ContextStringIO(BytesIO):
+    """
+    Wrapper to allow use of StringIO via 'with' constructs.
+
+    """
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        self.close()
+        return False
+
+
+class JsonSerializer(Serializer):
+    """
+    JsonSerializer serializes objects in the Confluent Schema Registry binary
+    format for JSON.
+
+    Note:
+        The ``title`` annotation, referred to as a record name
+        elsewhere in this document, is not strictly required by the JSON Schema
+        specification. It is however required by this Serializer. This
+        annotation(record name) is used to register the Schema with the Schema
+        Registry. See documentation below  for additional details on Subjects
+        and schema registration.
+
+    JsonSerializer configuration properties:
+    +-----------------------+----------+--------------------------------------------------+
+    | Property Name         | type     | Description                                      |
+    +=======================+==========+==================================================+
+    |                       |          | Registers schemas automatically if not           |
+    | auto.register.schemas | bool     | previously associated with a particular subject. |
+    |                       |          | Defaults to True.                                |
+    +-----------------------|----------+--------------------------------------------------+
+    |                       |          | Callable(str, SerialalizationContext) -> str     |
+    |                       |          |                                                  |
+    | subject.name.strategy | callable | Instructs the JsonSerializer on how to Construct |
+    |                       |          | Schema Registry subject names.                   |
+    |                       |          | Defaults to topic_subject_name_strategy.         |
+    +-----------------------+----------+--------------------------------------------------+
+
+    Schemas are registered to namespaces known as Subjects which define how a
+    schema may evolve over time. By default the subject name is formed by
+    concatenating the topic name with the message field separated by a hyphen.
+
+    i.e. {topic name}-{message field}
+
+    Alternative naming strategies may be configured with the property
+    `subject.name.strategy`.
+
+    Supported subject name strategies:
```

Comment:
maybe "Subject name strategies supported out of the box:" (you support more than just these three!)

Reply:
In addition to these three we support a reference name strategy which is technically supported but only has one implementation. For now I think this is sufficient. Let's revisit this in a follow-up docs PR I'm planning to verify all the rendering.

Reply:
reference name strategy is arguably a different type of thing, and this text still misleading  (but minor thing)